core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -119,33 +123,55 @@ func (ec *ethConfirmer) CheckForReceipts() error {
 	if len(unconfirmedEtxs) > 0 {
 		logger.Debugf("EthConfirmer: %v unconfirmed transactions", len(unconfirmedEtxs))
 	}
+	wg := sync.WaitGroup{}
+	wg.Add(receiptFetcherWorkerCount)
+	chEthTxes := make(chan models.EthTx)
+	for i := 0; i < receiptFetcherWorkerCount; i++ {
+		go ec.fetchReceipts(chEthTxes, &wg)
+	}
 	for _, etx := range unconfirmedEtxs {
+		chEthTxes <- etx
+	}
+	close(chEthTxes)
+	wg.Wait()
+	return nil
+}
+
+func (ec *ethConfirmer) fetchReceipts(chEthTxes <-chan models.EthTx, wg *sync.WaitGroup) {
+	defer wg.Done()
+	for {
+		etx, ok := <-chEthTxes
+		if !ok {
+			return
+		}
 		for _, attempt := range etx.EthTxAttempts {
-			// NOTE: If this becomes a performance bottleneck due to eth node requests,
-			// it may be possible to use goroutines here to speed it up by
-			// issuing `fetchReceipt` requests in parallel
+			// NOTE: This could conceivably be optimised even further at the
+			// expense of slightly higher load for the remote eth node, by
+			// batch requesting all receipts at once
```

Comment:
Would vote for this.  I can help you implement if you want as I did something similar for the HeadTracker (temporarily, didn't make it into `develop`)